src/mpl/src/gavl/mpl_gavl.c

```diff
@@ -289,12 +503,26 @@ int MPL_gavl_tree_search(MPL_gavl_tree_t gavl_tree, const void *addr, uintptr_t
     while (cur_node) {
         int cmp_ret = gavl_subset_cmp_func((uintptr_t) addr, len, cur_node);
         if (cmp_ret == BUFFER_MATCH) {
-            *val = (void *) cur_node->val;
+            gavl_tree_subnode_s *sub_node;
+            sub_node = cur_node->subnode_list;
+            assert(sub_node != NULL);
+
+            do {
+                cmp_ret = gavl_subset_subnode_cmp_func((uintptr_t) addr, len, sub_node);
```

Comment:
How do you order the buffers in the subnode list?

Reply:
there is no order in subnode list.